### PR TITLE
[testbed] disable setting grpc log

### LINKED ISSUE #9557: Failing tests: TestHarness_MetricsGoldenData/otlp-otlp
This test has been failing intermittently, see example:

https://github.com/open-telemetry/opentelemetry-collector-contrib/runs/6196786412?check_suite_focus=true

```
==================
WARNING: DATA RACE
Write at 0x000006d82c10 by goroutine 22:
  google.golang.org/grpc/grpclog.SetLoggerV2()
      /home/runner/go/pkg/mod/google.golang.org/grpc@v1.46.0/grpclog/loggerv2.go:76 +0x11a
  go.opentelemetry.io/collector/service/internal/telemetrylogs.SetColGRPCLogger()
      /home/runner/go/pkg/mod/go.opentelemetry.io/collector@v0.49.1-0.20220426224620-b80d75452365/service/internal/telemetrylogs/logger.go:76 +0xf6
  go.opentelemetry.io/collector/service.(*Collector).setupConfigurationComponents()
      /home/runner/go/pkg/mod/go.opentelemetry.io/collector@v0.49.1-0.20220426224620-b80d75452365/service/collector.go:208 +0x3f2
  go.opentelemetry.io/collector/service.(*Collector).Run()
      /home/runner/go/pkg/mod/go.opentelemetry.io/collector@v0.49.1-0.20220426224620-b80d75452365/service/collector.go:249 +0x2dd
  github.com/open-telemetry/opentelemetry-collector-contrib/testbed/testbed.(*inProcessCollector).Start.func1()
      /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/testbed/testbed/in_process_collector.go:96 +0xc4

Previous read at 0x000006d82c10 by goroutine 21:
  google.golang.org/grpc/internal/grpclog.InfoDepth()
      /home/runner/go/pkg/mod/google.golang.org/grpc@v1.46.0/internal/grpclog/grpclog.go:37 +0x93
  google.golang.org/grpc/grpclog.(*componentData).InfoDepth()
      /home/runner/go/pkg/mod/google.golang.org/grpc@v1.46.0/grpclog/component.go:36 +0x173
  google.golang.org/grpc/internal/channelz.AddTraceEvent()
      /home/runner/go/pkg/mod/google.golang.org/grpc@v1.46.0/internal/channelz/funcs.go:340 +0x402
  google.golang.org/grpc/internal/channelz.Info()
      /home/runner/go/pkg/mod/google.golang.org/grpc@v1.46.0/internal/channelz/logging.go:35 +0xb3
  google.golang.org/grpc.(*Server).Serve()
      /home/runner/go/pkg/mod/google.golang.org/grpc@v1.46.0/server.go:776 +0x486
  go.opentelemetry.io/collector/receiver/otlpreceiver.(*otlpReceiver).startGRPCServer.func1()
      /home/runner/go/pkg/mod/go.opentelemetry.io/collector@v0.49.1-0.20220426224620-b80d75452365/receiver/otlpreceiver/otlp.go:80 +0xd8

Goroutine 22 (running) created at:
  github.com/open-telemetry/opentelemetry-collector-contrib/testbed/testbed.(*inProcessCollector).Start()
      /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/testbed/testbed/in_process_collector.go:94 +0x648
  github.com/open-telemetry/opentelemetry-collector-contrib/testbed/correctnesstests/metrics.(*correctnessTestCase).startCollector()
      /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/testbed/correctnesstests/metrics/correctness_test_case.go:57 +0x53e
  github.com/open-telemetry/opentelemetry-collector-contrib/testbed/correctnesstests/metrics.testWithMetricsGoldenDataset()
      /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/testbed/correctnesstests/metrics/metrics_correctness_test.go:72 +0x3be
  github.com/open-telemetry/opentelemetry-collector-contrib/testbed/correctnesstests/metrics.TestHarness_MetricsGoldenData.func1()
      /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/testbed/correctnesstests/metrics/metrics_correctness_test.go:44 +0xac
  testing.tRunner()
      /opt/hostedtoolcache/go/1.18.0/x64/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /opt/hostedtoolcache/go/1.18.0/x64/src/testing/testing.go:1486 +0x47

Goroutine 21 (running) created at:
  go.opentelemetry.io/collector/receiver/otlpreceiver.(*otlpReceiver).startGRPCServer()
      /home/runner/go/pkg/mod/go.opentelemetry.io/collector@v0.49.1-0.20220426224620-b80d75452365/receiver/otlpreceiver/otlp.go:77 +0x1eb
  go.opentelemetry.io/collector/receiver/otlpreceiver.(*otlpReceiver).startProtocolServers()
      /home/runner/go/pkg/mod/go.opentelemetry.io/collector@v0.49.1-0.20220426224620-b80d75452365/receiver/otlpreceiver/otlp.go:127 +0x50d
  go.opentelemetry.io/collector/receiver/otlpreceiver.(*otlpReceiver).Start()
      /home/runner/go/pkg/mod/go.opentelemetry.io/collector@v0.49.1-0.20220426224620-b80d75452365/receiver/otlpreceiver/otlp.go:155 +0x44
  go.opentelemetry.io/collector/internal/sharedcomponent.(*SharedComponent).Start.func1()
      /home/runner/go/pkg/mod/go.opentelemetry.io/collector@v0.49.1-0.20220426224620-b80d75452365/internal/sharedcomponent/sharedcomponent.go:74 +0x8d
  sync.(*Once).doSlow()
      /opt/hostedtoolcache/go/1.18.0/x64/src/sync/once.go:68 +0x101
  sync.(*Once).Do()
      /opt/hostedtoolcache/go/1.18.0/x64/src/sync/once.go:59 +0x46
  go.opentelemetry.io/collector/internal/sharedcomponent.(*SharedComponent).Start()
      /home/runner/go/pkg/mod/go.opentelemetry.io/collector@v0.49.1-0.20220426224620-b80d75452365/internal/sharedcomponent/sharedcomponent.go:73 +0xc8
  github.com/open-telemetry/opentelemetry-collector-contrib/testbed/testbed.(*BaseOTLPDataReceiver).Start()
      /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/testbed/testbed/receivers.go:110 +0xb33
  github.com/open-telemetry/opentelemetry-collector-contrib/testbed/correctnesstests/metrics.(*correctnessTestCase).startTestbedReceiver()
      /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/testbed/correctnesstests/metrics/correctness_test_case.go:77 +0x14b
  github.com/open-telemetry/opentelemetry-collector-contrib/testbed/correctnesstests/metrics.testWithMetricsGoldenDataset()
      /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/testbed/correctnesstests/metrics/metrics_correctness_test.go:71 +0x3b1
  github.com/open-telemetry/opentelemetry-collector-contrib/testbed/correctnesstests/metrics.TestHarness_MetricsGoldenData.func1()
      /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/testbed/correctnesstests/metrics/metrics_correctness_test.go:44 +0xac
  testing.tRunner()
      /opt/hostedtoolcache/go/1.18.0/x64/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /opt/hostedtoolcache/go/1.18.0/x64/src/testing/testing.go:1486 +0x47
==================
```

### PR DESCRIPTION
**Pull request by codeboten**
This is causing a race condition with the InProcessCollector.

This change includes:
- set the flag to disable setting the grpc logger from the `InProcessCollector`
- `make update-otel`
- update calls to `featuregate` as a result of https://github.com/open-telemetry/opentelemetry-collector/commit/addf0ef351aecf9d944c09ff45d1c7d7d45e265a
- update tests as a result of https://github.com/open-telemetry/opentelemetry-collector/commit/5658f35067382cf5213ee2538a01ab66fa2ca084

Fixes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/9557

### CODE REVIEW

File: testbed/testbed/in_process_collector.go [hunk 1]
```diff
@@ -80,9 +80,10 @@ func (ipp *inProcessCollector) Start(args StartParams) error {
 	}
 
 	settings := service.CollectorSettings{
-		BuildInfo:      component.NewDefaultBuildInfo(),
-		Factories:      ipp.factories,
-		ConfigProvider: configProvider,
+		BuildInfo:             component.NewDefaultBuildInfo(),
+		Factories:             ipp.factories,
+		ConfigProvider:        configProvider,
+		SkipSettingGRPCLogger: true,
```
Comment (jpkrohling) [inline testbed/testbed/in_process_collector.go:86]: Now it all makes sense :-)

**Review by codeboten — COMMENTED** [review]
@bogdandrutu done

**Comment by bogdandrutu** [comment]
```console
=== RUN   TestMetricGroupData_toDistributionUnitTest/histogram_with_startTimestamp_of_11
    otlp_metricfamily_test.go:181: 
        	Error Trace:	otlp_metricfamily_test.go:181
        	Error:      	Not equal: 
        	            	expected: internal.HistogramDataPoint{orig:(*v1.HistogramDataPoint)(0xc00012e360)}
        	            	actual  : internal.HistogramDataPoint{orig:(*v1.HistogramDataPoint)(0xc00012e2d0)}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -29,4 +29,3 @@
        	            	   },
        	            	-  ExplicitBounds: ([]float64) {
        	            	-  },
        	            	+  ExplicitBounds: ([]float64) <nil>,
        	            	   Exemplars: ([]v1.Exemplar) <nil>,
        	Test:       	TestMetricGroupData_toDistributionUnitTest/histogram_with_startTimestamp_of_11
        	Messages:   	Expected the points to be equal
=== RUN   TestMetricGroupData_toDistributionUnitTest/histogram_that_is_stale
    otlp_metricfamily_test.go:181: 
        	Error Trace:	otlp_metricfamily_test.go:181
        	Error:      	Not equal: 
        	            	expected: internal.HistogramDataPoint{orig:(*v1.HistogramDataPoint)(0xc0003f3290)}
        	            	actual  : internal.HistogramDataPoint{orig:(*v1.HistogramDataPoint)(0xc0003f3200)}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -27,4 +27,3 @@
        	            	   },
        	            	-  ExplicitBounds: ([]float64) {
        	            	-  },
        	            	+  ExplicitBounds: ([]float64) <nil>,
        	            	   Exemplars: ([]v1.Exemplar) <nil>,
        	Test:       	TestMetricGroupData_toDistributionUnitTest/histogram_that_is_stale
        	Messages:   	Expected the points to be equal
```

This error is caused by a change in `CopyTo` where we don't copy empty array, we live it nil in that case.

**Comment by dmitryax** [comment]
@codeboten @bogdandrutu do you think we can update the test to change original slices to nil? I can also update the CopyTo to make exact copy of slices

**Comment by codeboten** [comment]
@dmitryax I updated the test

File: exporter/googlecloudexporter/config_test.go [hunk 2]
```diff
@@ -32,10 +32,10 @@ import (
 // setPdataFeatureGateForTest changes the pdata feature gate during a test.
 // usage: defer SetPdataFeatureGateForTest(true)()
 func setPdataFeatureGateForTest(enabled bool) func() {
-	originalValue := featuregate.IsEnabled(pdataExporterFeatureGate)
-	featuregate.Apply(map[string]bool{pdataExporterFeatureGate: enabled})
+	originalValue := featuregate.GetRegistry().IsEnabled(pdataExporterFeatureGate)
```
Comment (tigrannajaryan) [inline exporter/googlecloudexporter/config_test.go:35]: I understand that the featuregate-related changes are forced, but the commit message does not match it. Perhaps rewrite the commit message to make it more relevant.
The actual change that matches the commit message is drowned in the featuregate changes.

**Review by tigrannajaryan — APPROVED** [review]

File: exporter/googlecloudexporter/config_test.go (hunk 2 above)
Comment (codeboten) [reply exporter/googlecloudexporter/config_test.go:35]: @tigrannajaryan I'll add a note to the merge commit if that works for you, i'd rather not wait for another build if you're ok with this.

**Review by dmitryax — APPROVED** [review]

File: exporter/googlecloudexporter/config_test.go (hunk 2 above)
Comment (codeboten) [reply exporter/googlecloudexporter/config_test.go:35]: see the merge commit message: https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/b9915f749b7ec03fabcff3200bc659e3fe9f549d

Comment (tigrannajaryan) [reply exporter/googlecloudexporter/config_test.go:35]: Thanks!

**Pull request merged**